src/core/lib/layer.js

```diff
@@ -495,16 +495,18 @@ export default class Layer {
 
     const propsChanged = Boolean(propsChangedReason);
     const dataChanged = Boolean(dataChangedReason);
-    const propsOrDataChanged = propsChanged || dataChanged;
     const viewportChanged = context.viewportChanged;
-    const somethingChanged = propsChanged || dataChanged || viewportChanged;
 
+    let updateTriggersChanged = false;
     // Check update triggers to determine if any attributes need regeneration
     // Note - if data has changed, all attributes will need regeneration, so skip this step
     if (!dataChanged) {
-      this._diffUpdateTriggers(oldProps, newProps);
+      updateTriggersChanged = this._diffUpdateTriggers(oldProps, newProps);
     }
 
+    const propsOrDataChanged = propsChanged || dataChanged || updateTriggersChanged;
```

Comment:
Q: Do we need to set `propsOrDataChanged`? If we set `propsOrDataChanged` to `true`,  I believe that we will also cause layers to update props. Not a terrible thing, but still unnecessary work in cases where it was just an `updateTrigger` that fired?

Would it make sense to just use the new flag `changeFlags.updateTriggersChanged` and trigger attribute invalidation based on that?

Reply:
`diffUpdateTrigger` already invalidates the attribute, the problem is that it doesn't recalculate them.
Invalidated attributes are updated in `updateState`, which is blocked by `propsOrDataChanged`.